patches/minecraft/net/minecraft/entity/passive/HorseArmorType.java.patch

```diff
@@ -0,0 +1,65 @@
+--- ../src-base/minecraft/net/minecraft/entity/passive/HorseArmorType.java
++++ ../src-work/minecraft/net/minecraft/entity/passive/HorseArmorType.java
+@@ -24,7 +24,7 @@
+         this.field_188586_e = null;
+         this.field_188587_f = "";
+     }
+-
++    
+     private HorseArmorType(int p_i46800_3_, String p_i46800_4_, String p_i46800_5_)
+     {
+         this.field_188588_g = p_i46800_3_;
+@@ -55,6 +55,7 @@
+         return this.field_188586_e;
+     }
+ 
++    @Deprecated
+     public static HorseArmorType func_188575_a(int p_188575_0_)
+     {
+         return values()[p_188575_0_];
+@@ -62,9 +63,14 @@
+ 
+     public static HorseArmorType func_188580_a(ItemStack p_188580_0_)
+     {
+-        return p_188580_0_.func_190926_b() ? NONE : func_188576_a(p_188580_0_.func_77973_b());
++        if(p_188580_0_.func_77973_b() instanceof net.minecraftforge.common.IHorseArmor)
++        {
++            return ((net.minecraftforge.common.IHorseArmor) p_188580_0_.func_77973_b()).getHorseArmorType(p_188580_0_);
++        }
++        return p_188580_0_.func_77973_b() == Items.field_151138_bX ? net.minecraft.entity.passive.HorseArmorType.IRON : p_188580_0_.func_77973_b() == Items.field_151136_bY ? net.minecraft.entity.passive.HorseArmorType.GOLD : p_188580_0_.func_77973_b() == Items.field_151125_bZ ? net.minecraft.entity.passive.HorseArmorType.DIAMOND : net.minecraft.entity.passive.HorseArmorType.NONE;
+     }
+ 
++    @Deprecated
+     public static HorseArmorType func_188576_a(Item p_188576_0_)
+     {
+         if (p_188576_0_ == Items.field_151138_bX)
+@@ -81,8 +87,29 @@
+         }
+     }
+ 
++    @Deprecated
```

Comment:
put a comment saying what method to use instead:
`@Deprecated // Forge: use ___ method instead`